src/request/sign-transaction/SignTransactionApi.js

```diff
@@ -23,14 +24,29 @@ class SignTransactionApi extends TopLevelApi {
         parsedRequest.recipientLabel = this.parseLabel(request.recipientLabel);
         parsedRequest.transaction = this.parseTransaction(request);
         parsedRequest.layout = this.parseLayout(request.layout);
-        if (parsedRequest.layout === SignTransactionApi.Layouts.CHECKOUT) {
+        if (parsedRequest.layout === SignTransactionApi.Layouts.CHECKOUT
+            && request.layout === SignTransactionApi.Layouts.CHECKOUT) {
             parsedRequest.shopOrigin = this.parseShopOrigin(request.shopOrigin);
             parsedRequest.shopLogoUrl = this.parseShopLogoUrl(request.shopLogoUrl);
             if (parsedRequest.shopLogoUrl && parsedRequest.shopLogoUrl.origin !== parsedRequest.shopOrigin) {
                 throw new Errors.InvalidRequestError('origin of shopLogoUrl must be same as referrer');
             }
-        } else {
-            parsedRequest.shopOrigin = undefined;
+
+            parsedRequest.fiatAmount = this.parseNonNegativeFiniteNumber(request.fiatAmount);
+            parsedRequest.fiatCurrency = this.parseFiatCurrency(request.fiatCurrency);
+            if ((parsedRequest.fiatAmount === undefined) !== (parsedRequest.fiatCurrency === undefined)) {
+                throw new Errors.InvalidRequestError('fiatAmount and fiatCurrency must be both defined or undefined.');
+            }
+
+            parsedRequest.time = this.parseNonNegativeFiniteNumber(request.time);
+            parsedRequest.expires = this.parseNonNegativeFiniteNumber(request.expires);
+            if (parsedRequest.expires !== undefined) {
+                if (parsedRequest.time === undefined) {
```

Comment:
`time` and `expires` should never be `0`, right? So all these tests for `undefined` specifically are unnecessary IMO.

Reply:
In this case typescript can't infer that in line 46 `parsedRequest.time` can be `undefined` if not checking for `undefined` here.
In other cases it's essentially a matter of taste.

Reply:
For me it's not just about taste, but also about defensive programming. Just because we tell Typescript that it can be `string | undefined` (for example), does not mean that outside calls cannot pass `null` or other types in. Typescript does not do type checks during runtime, and when it's not checked by whitelist anywhere, types are passed through as they are. So to be sure, I would always do a falsy or type whitelist check for formerly untested arguments.